File: HeadFirstDesignPatternsOC/StrategyPattern/Duck/ModelDuck.m
//
//  ModelDuck.m
//  StrategyPattern
//
//  Created by 陳峻琦 on 25/5/2022.
//

#import "ModelDuck.h"
#import "FlyNoWay.h"
#import "Quack.h"

@implementation ModelDuck

- (instancetype)init {
    if (self = [super init]) {
        self.flyBehavior = [FlyNoWay new];
        self.quackBehavior = [Quack new];
    }
    return self;
}

- (void)display {
    NSLog(@"I'm a model duck");
}

@end

File: HeadFirstDesignPatternsJava/HeadFirst/src/StrategyPattern/Duck/ModelDuck.java
package StrategyPattern.Duck;

import StrategyPattern.Fly.FlyNoWay;
import StrategyPattern.Quack.Quack;

public class ModelDuck extends Duck {

    public ModelDuck() {
        super();
        flyBehavior = new FlyNoWay();
        quackBehavior = new Quack();
    }

    @Override
    public void display() {
        System.out.println("I'm a model duck");
    }
    
}
